app/src/main/java/net/aiscope/gdd_app/ui/metadata/MetadataPresenter.kt

```diff
@@ -32,12 +32,7 @@ class MetadataPresenter @Inject constructor(
     suspend fun showScreen() {
         val sample = repository.current()
 
-        if (sample.disease == null) {
-            notValid()
-            return
-        }
-
-        val lastMetadata = repository.last()?.metadata
+        val lastMetadata = repository.lastSaved()?.metadata
```

Comment:
Makes me so happy to see null checks removed. 
Btw, with this occasion, the `notValid()` can be removed, as well as the `showInvalidFormError()` on the view, the string resource for the error message, and the disabled test (`shouldShowErrorMessageIfNoBloodTypeSelected()`) :)

Reply:
I love removing code 💅